### DISCUSSION TITLE
Way to cleanup Repository of Query Deleted Entities

### DISCUSSION THREAD

**Discussion by Hanmac** [doctrine/doctrine2#9908] [Support Questions]
Hi i have some Query made with Query Builder where i mass delete some entities where their id is *not* a certain value (plus other condition)

now i have the problem that the Repository, the EntityManager and the UnitOfWork does still remember the old entities, so the cheep actions like refresh or find by id doesn't work.

Now i would like a way for ORM to better refresh and forget the deleted entities.

**Comment by derrabus** [comment]
Clear the entity manager.

**Reply by Hanmac** [reply]
but wouldn't call the entityManager->clear more problems with the still existing entities not be found anymore?

**Reply by Hanmac** [reply]
i think i had problems with my test case when clearing the entityManager, i probably need to disable Proxy or something

**Reply by derrabus** [reply]
Yes, clearing the EM would detach all entities. But after such an operation, it is the reasonable thing to do.
